rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMController.java

```diff
@@ -1337,7 +1365,59 @@ public void onFailure(Throwable caught) {
 
                                  public void onSuccess(String result) {
-                                     syncCallBack.onSuccess(parseScriptResult(result));
+                                     syncCallBack.onSuccess(parseListScriptResult(result));
+                                 }
+                             });
+    }
+
+    public void executeNodeSourceScript(final String script, final String engine, final String nodeSourceName,
+            final Callback<String, String> syncCallBack) {
+        rm.executeNodeSourceScript(LoginModel.getInstance().getSessionId(),
+                                   script,
+                                   engine,
+                                   nodeSourceName,
+                                   new AsyncCallback<String>() {
+                                       public void onFailure(Throwable caught) {
+                                           String msg = JSONUtils.getJsonErrorMessage(caught);
+                                           LogModel.getInstance()
+                                                   .logImportantMessage("Failed to execute a script " + script +
+                                                                        " on node source " + nodeSourceName + " : " +
+                                                                        msg);
+                                           if (msg.equals("HTTP 500 Internal Server Error")) {
+                                               syncCallBack.onFailure("You are not authorized to execute scripts on this node source. Please contact the administrator.");
+                                           } else {
+                                               syncCallBack.onFailure(msg);
+                                           }
+
+                                       }
+
+                                       public void onSuccess(String result) {
+                                           syncCallBack.onSuccess(parseListScriptResult(result));
+                                       }
+                                   });
+    }
+
+    public void executeHostScript(final String script, final String engine, final String host,
+            final Callback<String, String> syncCallBack) {
+        rm.executeHostScript(LoginModel.getInstance().getSessionId(),
+                             script,
+                             engine,
+                             host,
+                             new AsyncCallback<String>() {
+                                 public void onFailure(Throwable caught) {
+                                     String msg = JSONUtils.getJsonErrorMessage(caught);
+                                     LogModel.getInstance().logImportantMessage("Failed to execute a script " + script +
+                                                                                " on host " + host + " : " + msg);
+                                     if (msg.equals("HTTP 500 Internal Server Error")) {
```

Comment:
you can then reuse the handleError method as before

Reply:
Yes.